tests/src/test/scala/cats/tests/MiniFloatSuite.scala

```diff
@@ -0,0 +1,393 @@
+      val miniFloat = MiniFloat.from(float)
+
+      // This is necessary since Float.NaN != Float.NaN
+      if (miniFloat.isNaN) {
+        assert(expected.isNaN)
+      } else {
+        assert(miniFloat.toFloat === expected)
+      }
+    }
+
+  testFloatConversion(Float.NegativeInfinity, MiniFloat.NegativeInfinity)
+  testFloatConversion(Float.MinValue, MiniFloat.NegativeInfinity)
+  testFloatConversion(-16f, MiniFloat.NegativeInfinity)
+  testFloatConversion(Math.nextDown(-12f), MiniFloat.NegativeInfinity)
+  testFloatConversion(-12f, -8f)
+  testFloatConversion(Math.nextDown(-6f), -8f)
+  testFloatConversion(-6f, -4f)
+  testFloatConversion(Math.nextDown(-3f), -4f)
+  testFloatConversion(-3f, -2f)
+  testFloatConversion(-1f, MiniFloat.NegativeOne)
+  testFloatConversion(1f, MiniFloat.One)
+  testFloatConversion(0f, MiniFloat.Zero)
+  testFloatConversion(Float.MinPositiveValue, MiniFloat.Zero)
+  testFloatConversion(-0f, MiniFloat.Zero)
+  testFloatConversion(Math.nextDown(0.125f), MiniFloat.Zero)
+  testFloatConversion(0.125f, MiniFloat.Zero)
+  testFloatConversion(Math.nextDown(0.25f), MiniFloat.Zero)
+  testFloatConversion(0.25f, MiniFloat.Zero)
+  testFloatConversion(0.5f, 0.5f)
+  testFloatConversion(1f, 1f)
+  testFloatConversion(2f, 2f)
+  testFloatConversion(Math.nextDown(3f), 2f)
+  testFloatConversion(3f, 4f)
+  testFloatConversion(4f, 4f)
+  testFloatConversion(5f, 4f)
+  testFloatConversion(6f, 8f)
+  testFloatConversion(7f, 8f)
+  testFloatConversion(8f, 8f)
+  testFloatConversion(Math.nextDown(12f), 8f)
+  testFloatConversion(12f, MiniFloat.PositiveInfinity)
+  testFloatConversion(16f, MiniFloat.PositiveInfinity)
+  testFloatConversion(Float.MaxValue, MiniFloat.PositiveInfinity)
+  testFloatConversion(Float.PositiveInfinity, MiniFloat.PositiveInfinity)
+  testFloatConversion(Float.NaN, MiniFloat.NaN)
+
+  test("MiniFloat.fromFloat(-0) is not negative") {
+    assert(!(MiniFloat.from(-0).toFloat < 0))
+  }
+
+  test(s"fromDouble consistent with fromFloat") {
+    forAll { (n: Double) =>
+      assert(MiniFloat.from(n) === MiniFloat.from(n.toFloat), n)
+    }
+  }
+
+  test(s"toDouble consistent with toFloat") {
+    forAll { (mf: MiniFloat) =>
+      val fromToDouble = mf.toDouble.toFloat
+      val fromToFloat = mf.toFloat
+
+      // This is necessary since Float.NaN != Float.NaN
+      if (fromToDouble.isNaN) {
+        assert(fromToFloat.isNaN)
+      } else {
+        assert(fromToDouble === fromToFloat, mf)
+      }
+    }
+  }
+
+  // Special number tests and behaviours
+
+  private def testSpecialNumberExpectations(
+    mf: MiniFloat,
+    expectIsNaN: Boolean,
+    expectIsFinite: Boolean
+  )(implicit
+    loc: Location
+  ): Unit = {
+    test(s"$mf isNan $expectIsNaN") {
+      assert(mf.isNaN === expectIsNaN)
+    }
+
+    test(s"$mf isFinite $expectIsFinite") {
+      assert(mf.isFinite === expectIsFinite)
+    }
+  }
+
+  testSpecialNumberExpectations(MiniFloat.NaN, expectIsNaN = true, expectIsFinite = false)
```

Comment:
This test fails in Scala Native due to https://github.com/scala-native/scala-native/issues/2178.